integration_tests/src/main/python/map_test.py

```diff
@@ -32,6 +38,44 @@ def test_simple_get_map_value(data_gen):
                 'a["NOT_FOUND"]',
                 'a["key_5"]'))
 
+def test_map_expr_project_string_to_string():
+    data_gen = [('a', StringGen(nullable=False)), ('b', StringGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+def test_map_expr_project_int_to_int():
+    data_gen = [('a', IntegerGen(nullable=False)), ('b', IntegerGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m2"))
+
+def test_map_expr_project_string_to_int():
+    data_gen = [('a', StringGen(nullable=False)), ('b', IntegerGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+def test_map_expr_project_int_to_string():
+    data_gen = [('a', IntegerGen(nullable=False)), ('b', StringGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+def test_map_expr_project_struct_to_struct():
+    data_gen = [('a', basic_struct_gen), ('b', basic_struct_gen)]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+@allow_non_gpu('ProjectExec,Alias,CreateMap')
+# until https://github.com/NVIDIA/spark-rapids/issues/3229 is implemented
+def test_map_expr_multiple_pairs():
+    data_gen = [('a', StringGen(nullable=False)), ('b', StringGen(nullable=False))]
+    assert_gpu_fallback_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b, b, a) as m1"), 'ProjectExec')
```

Comment:
What happens if "a == b" for a given row?  

Spark has a config for what should happen.

https://github.com/apache/spark/blob/f620996142ba312f7e52f75476b1b18be667ffdf/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L2969-L2982

Sadly neither of those we are going to be able to support out of the box. I think we need to file a follow on issue so that we can figure out how to dedupe maps the proper way, with cudf's help. In the short term we might need to have a config indicating that it is okay to enable this because we are not doing duplicate checks, but if the keys are all scalar values we can determine duplicates up front and avoid the entire issue.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/3229 is the follow-on issue for supporting multiple key-value pairs and where we would need to tackle the duplicate issue.

Reply:
It can be a separate issue after that or as a part of it. I think most of the time people will use literal values for the keys. If that is the case, then we can still enable this for those cases without much difficulty, but we would need to fall back to the CPU for other cases, or have a config to say I know what I am doing.